server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -840,6 +856,46 @@ private static void ccsClusterInfoUpdate(
         } while (swapped == false);
     }
 
+    /**
+     * Edge case ---
+     * Typically we don't need to update a Cluster object after the SearchShards API call, since the
+     * skipped shards will be passed into SearchProgressListener.onListShards.
+     * However, there are edge cases where the remote SearchShards API call returns no shards at all.
+     * So in that case, nothing for this cluster will be passed to onListShards, so we need to update
+     * the Cluster object to SUCCESSFUL status with shard counts of 0 and a filled in 'took' value.
+     *
+     * @param response from SearchShards API call to remote cluster
+     * @param clusterRef Reference Cluster to be updated
+     * @param timeProvider search time provider (for setting took value)
+     */
+    private static void ccsClusterInfoUpdate(
+        SearchShardsResponse response,
+        AtomicReference<SearchResponse.Cluster> clusterRef,
+        SearchTimeProvider timeProvider
+    ) {
+        if (response.getGroups().isEmpty()) {
+            boolean swapped;
+            do {
+                SearchResponse.Cluster orig = clusterRef.get();
+                SearchResponse.Cluster updated = new SearchResponse.Cluster(
+                    orig.getClusterAlias(),
+                    orig.getIndexExpression(),
+                    orig.isSkipUnavailable(),
+                    SearchResponse.Cluster.Status.SUCCESSFUL,
```

Comment:
I'm not totally sure what the Cluster search Status should be here, as I don't understand why this edge case occurs (I was not able to reproduce it in my IT tests). 

It was uncovered by the existing CrossClusterSearchUnavailableClusterIT test [on this line](https://github.com/elastic/elasticsearch/blob/main/qa/ccs-unavailable-clusters/src/javaRestTest/java/org/elasticsearch/search/CrossClusterSearchUnavailableClusterIT.java#L206).

I've read over that test and I honestly don't understand what is going in that test. Since `skip_unavailable=false` for that part of the test, I would have thought the search should fail rather than mark the cluster with no shards to search as "successful", but since that is existing functionality, I modeled this code after that.

I'm open to change.